src/librustc/ty/layout.rs

```diff
@@ -2301,7 +2301,7 @@ impl<'tcx> ty::Instance<'tcx> {
     // or should go through `FnAbi` instead, to avoid losing any
     // adjustments `FnAbi::of_instance` might be performing.
     fn fn_sig_for_fn_abi(&self, tcx: TyCtxt<'tcx>) -> ty::PolyFnSig<'tcx> {
-        let ty = self.ty(tcx);
+        let ty = self.monomorphic_ty(tcx);
```

Comment:
It's not obvious to me that this is correct. Ca you go up the call chain and see if there are any noncodegen uses?

Reply:
The only caller is `FnAbi:of_instance`, which is only called from codegen.

Reply:
We need to stop assuming that codegen = monomorphic.